Sources/UIKitBackend/UIKitBackend+Path.swift

```diff
@@ -0,0 +1,177 @@
+                case .lineTo(let point):
+                    path.addLine(to: CGPoint(x: point.x, y: point.y))
+                case .quadCurve(let control, let end):
+                    path.addQuadCurve(
+                        to: CGPoint(x: end.x, y: end.y),
+                        controlPoint: CGPoint(x: control.x, y: control.y)
+                    )
+                case .cubicCurve(let control1, let control2, let end):
+                    path.addCurve(
+                        to: CGPoint(x: end.x, y: end.y),
+                        controlPoint1: CGPoint(x: control1.x, y: control1.y),
+                        controlPoint2: CGPoint(x: control2.x, y: control2.y)
+                    )
+                case .rectangle(let rect):
+                    let cgPath: CGMutablePath = path.cgPath.mutableCopy()!
+                    cgPath.addRect(
+                        CGRect(x: rect.x, y: rect.y, width: rect.width, height: rect.height)
+                    )
+                    path.cgPath = cgPath
+                case .circle(let center, let radius):
+                    let cgPath: CGMutablePath = path.cgPath.mutableCopy()!
+                    cgPath.addEllipse(
+                        in: CGRect(
+                            x: center.x - radius,
+                            y: center.y - radius,
+                            width: radius * 2.0,
+                            height: radius * 2.0
+                        )
+                    )
+                    path.cgPath = cgPath
+                case .arc(let center, let radius, let startAngle, let endAngle, let clockwise):
+                    path.addArc(
+                        withCenter: CGPoint(x: center.x, y: center.y),
+                        radius: CGFloat(radius),
+                        startAngle: CGFloat(startAngle),
+                        endAngle: CGFloat(endAngle),
+                        clockwise: clockwise
+                    )
+                case .transform(let transform):
+                    path.apply(CGAffineTransform(transform))
+                case .subpath(let subpathActions):
+                    let subpath = UIBezierPath()
+                    applyActions(subpathActions, to: subpath)
+                    path.append(subpath)
+            }
+        }
+    }
+
+    public func renderPath(
+        _ path: Path,
+        container: Widget,
+        strokeColor: Color,
+        fillColor: Color,
+        overrideStrokeStyle: StrokeStyle?
+    ) {
+        if let overrideStrokeStyle {
+            applyStrokeStyle(overrideStrokeStyle, to: path)
+        }
+
+        let widget = container as! PathWidget
+        let shapeLayer = widget.shapeLayer
+
+        shapeLayer.path = path.cgPath
+        shapeLayer.lineWidth = path.lineWidth
+        shapeLayer.miterLimit = path.miterLimit
+        shapeLayer.fillRule = path.usesEvenOddFillRule ? .evenOdd : .nonZero
+
+        shapeLayer.lineJoin =
+            switch path.lineJoinStyle {
```

Comment:
The CI checks are warning about CGLineJoin and CGLineCap not being frozen enums. It's probably best to handle the `@unknown default:` case with a printed warning and a sensible default.

Reply:
If I add a print statement then this can no longer be a switch expression :/

but yeah, what you said makes sense